sgpt/app.py

```diff
@@ -71,7 +71,13 @@ def main(
         return
 
     if not prompt and not editor:
-        raise MissingParameter(param_hint="PROMPT", param_type="string")
+        if not sys.stdin.isatty():
+            prompt = sys.stdin.read()
+        else:
+            raise MissingParameter(param_hint="PROMPT", param_type="string")
+    elif prompt and not sys.stdin.isatty():
```

Comment:
Do we really need this second case?

If the user has provided a prompt, then why would we overwrite it with the standard input?

I think skipping this `elif` block also solves potential issues when stdin is not a TTY. (I removed it in my second patch.)